scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2014 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+# author:  Imron Alston <imron@scalyr.com>
+
+__author__ = 'imron@scalyr.com'
+
+import datetime
+import os
+import re
+import select
+from scalyr_agent import ScalyrMonitor, define_config_option
+from scalyr_agent.scalyr_monitor import BadMonitorConfiguration
+from scalyr_agent.monitor_utils import journald_checkpoints
+from scalyr_agent.monitor_utils.journald_checkpoints import Checkpoint
+import scalyr_agent.scalyr_logging as scalyr_logging
+
+try:
+    from systemd import journal
+except ImportError:
+    raise Exception('Python systemd library not installed.\n\nYou must install the systemd python library in order '
```

Comment:
No other places where it's raised (but several where it's caught).  That being said, the monitor modules are loaded dynamically, so it's already the case that the exception will only be raised if the user has configured the agent to use the journald monitor.  I just double checked this by setting it to always raise that exception and the error message only manifested when the journald_monitor was configured, otherwise the agent started normally.  I can make it throw in the initialize method if necessary, but the behaviour is already the same as you described so I'm not sure if it's necessary.

Reply:
Ok, we can leave it where it is as long as the error message is being shown at the right place.  Just add in a comment saying this should only be thrown during monitor configuration.